### PR TITLE
bugfix: pass `--ignore-platform-reqs` and `--ignore-platform-req` to `composer update --lock`

### LINKED ISSUE #28: Ignored platform requirements are not passed to lock update command
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 2.1.0

#### Summary

In case, platform requirements are being ignored, the composer v2 plugin does not properly pass those arguments to the lock-Update command.

#### Current behavior

When executing `composer update --prefer-lowest --prefer-stable --ignore-platform-req=php` on PHP 8.0, it is most likely the case, that a version of a package is being installed which is not compatible with PHP 8.0.

As of the base functionality of the composer v2 plugin, we have to update the `composer.lock` after the command mentioned above has finished.

The `composer update --lock` command needs those Arguments which are used to ignore platform requirements aswell.

#### How to reproduce

https://travis-ci.com/github/laminas/laminas-servicemanager/jobs/456178037#L639

#### Expected behavior

Downgrade works and installs `laminas/laminas-code` v3.3.1 even tho it does not support PHP 8.0.

### PR DESCRIPTION
**Pull request by boesing**
|    Q          |   A
|-------------- | ------
| Bugfix        | yes

### Description

Fixes #28

### CODE REVIEW

**Comment by boesing** [comment]
Relates to laminas/laminas-servicemanager#64

**Review by snapshotpl — APPROVED** [review]

File: src/DependencyRewriterV2.php [hunk 1]
```diff
@@ -58,6 +67,28 @@ public function __construct(?callable $applicationFactory = null, $composerFile
         };
     }
 
+    public function onPreCommandRun(PreCommandRunEvent $event)
+    {
+        parent::onPreCommandRun($event);
+        $input = $event->getInput();
+
+        $options = [];
+        foreach (self::COMPOSER_LOCK_UPDATE_OPTIONS as $optionName) {
```
Comment (boesing) [inline src/DependencyRewriterV2.php]: Reminder to myself:
Find another way to extract passed arguments to `composer` as this package may not be installed when command is being executed.

So there should be away of getting these arguments during runtime and not by hooking into this event.

Comment (weierophinney) [reply src/DependencyRewriterV2.php]: I don't understand your note to yourself. It sounds like you're worried about identifying those arguments when the plugin is not yet installed... are you thinking about a scenario where the plugin is listed in the application, but dependencies have never been installed, so Composer installs the plugins _first_, and then continues installing other dependencies?

In that case, doesn't Composer essentially do a second pass?

If it doesn't... you could always look at `$argv`, as it's a superglobal, and neither Composer nor symfony/console change the values it contains.

Comment (boesing) [reply src/DependencyRewriterV2.php]: > are you thinking about a scenario where the plugin is listed in the application, but dependencies have never been installed, so Composer installs the plugins first, and then continues installing other dependencies?

Yes, that was one of those scenarios I stumbled on when implementing composer v2 support.

> In that case, doesn't Composer essentially do a second pass?

No it doesn't. Only if the plugin was available when executing the command, the plugin is being called.
So yes, I was thinking about that `$argv` aswell or probably some OOP way. Need to verify the possibilities.

**Pull request merged**